types/datetime.luau

```diff
@@ -0,0 +1,180 @@
+-- TODO: Add more docs
+
+export type Locale = "en" | "de" | "es" | "fr" | "it" | "ja" | "pl" | "pt-br" | "pt" | "tr"
+
+export type Timezone = "utc" | "local"
+
+export type ShortMonth =
+	"jan"
+	| "feb"
+	| "mar"
+	| "apr"
+	| "may"
+	| "jun"
+	| "jul"
+	| "aug"
+	| "sep"
+	| "oct"
+	| "nov"
+	| "dec"
+
+export type Month =
+	"january"
+	| "february"
+	| "march"
+	| "april"
+	| "may"
+	| "june"
+	| "july"
+	| "august"
+	| "september"
+	| "october"
+	| "november"
+	| "december"
+
+export type DateTimeValues = {
+	year: number,
+	month: number | ShortMonth | Month,
```

Comment:
Seeing this in the type definitions will make users confused. Which one of the three types do the native methods actually return? Do I need to handle all of them as a consumer? We should probably just choose one and stick to it. I vote for an integer `1 -> 12` since it is consistent with the rest of the values.

Reply:
I mean we could create a union type with a different month field for functions which accept a `DateTimeValues` argument, and use `DateTimeValues` for the returned values.

Reply:
That could work, yeah. I think it's still worth considering if we should really have month as the only argument that diverges from being a `start >= N >= end` int range though

Reply:
I've gone ahead and made it so that the user can provide DateTimeValues with a month field that can be Month, ShortMonth or number, but the functions that return DateTimeValues with the month field of type number only.

See https://github.com/filiptibell/lune/pull/94/commits/2ca961544147191d68415da3a3061cdc7476e655.